File: frontend/src/microservices/content/objects/ComplaintCardPayload.ts
import type { FeedbackMessageType } from './FeedbackMessageType'
import type { FeedbackSenderRole } from './FeedbackSenderRole'

export type ComplaintMessageSnapshot = {
  messageId: string
  senderRole: FeedbackSenderRole
  senderDisplayName: string
  content: string
  messageType: FeedbackMessageType
  createdAt: string
}

export type ComplaintCardPayload = {
  complaintId: string
  sourceThreadId: string
  managerThreadId: string | null
  userExplanation: string
  summary: string
  targetDisplayName: string
  selectedMessages: ComplaintMessageSnapshot[]
  createdAt: string
}

File: backend/src/main/scala/microservices/feedback/objects/ComplaintCardPayload.scala
// ComplaintCardPayload：feedback 域后端对象定义。
package com.typesafe.travel.feedback.domain
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class ComplaintCardPayload(
    complaintId: String,
    sourceThreadId: String,
    managerThreadId: Option[String],
    userExplanation: String,
    summary: String,
    targetDisplayName: String,
    selectedMessages: List[ComplaintMessageSnapshot],
    createdAt: String
)
object ComplaintCardPayload:
  import ContentSourceJsonCodecs.given
  given sourceEncoder: Encoder[ComplaintCardPayload] = deriveEncoder
  given sourceDecoder: Decoder[ComplaintCardPayload] = deriveDecoder
